### PR TITLE
DE-4246 | api for getting bulk articles information

### PR DESCRIPTION
**Pull request by szczeles**

### CODE REVIEW

File: includes/wikia/api/CrossWikiArticlesApiController.class.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		// mapping parameters to wikiId => listOfArticles, and preparing cross-wiki sql
+		$wikiToArticleMap = array();
+		$sqlFilters = array();
+		foreach ( $articles as $article ) {
+			list( $wikiId, $articleId ) = explode('_', $article);
+			if ( ! isset($wikiToArticleMap[$wikiId]) ) {
+				$wikiToArticleMap[$wikiId] = array();
+			}
+			$wikiToArticleMap[$wikiId][] = $articleId;
+			$sqlFilters[] = sprintf('(page_wikia_id=%s and page_id=%s)', $wikiId, $articleId);
```
Comment (macbre) [inline includes/wikia/api/CrossWikiArticlesApiController.class.php]: Let's cast IDs here to integer to avoid SQL injection,

Comment (szczeles) [reply includes/wikia/api/CrossWikiArticlesApiController.class.php]: Thanks! I'm ashamed for introducing this SQL injection here 🙈

File: includes/wikia/api/CrossWikiArticlesApiController.class.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				'title' => $wikiDetails['title'],
+				'baseUrl' => $wikiDetails['url'],
+				'images' => $this->getThumbnails( $dbname, $wikiToArticleMap[$wikiId] )
+			);
+		}
+
+		// getting articles data
+		global $wgExternalDatawareDB;
+		$db = wfGetDB( DB_SLAVE, array(), $wgExternalDatawareDB );
+		$dbResult = $db->query( 'select * from pages where '. $multiArticleSqlFilter, __METHOD__ );
```
Comment (macbre) [inline includes/wikia/api/CrossWikiArticlesApiController.class.php]: Let's explicitly list columns we want to get from a database. This helps a lot when sunsetting no longer needed columns - we immediately know which columns are actually used.

File: includes/wikia/api/CrossWikiArticlesApiController.class.php (hunk 1 above)
Comment (macbre) [reply includes/wikia/api/CrossWikiArticlesApiController.class.php]: No worries :)

File: includes/wikia/api/CrossWikiArticlesApiController.class.php (hunk 2 above)
Comment (szczeles) [reply includes/wikia/api/CrossWikiArticlesApiController.class.php]: OK, changed 👍

**Review by macbre — APPROVED** [review]

**Pull request merged**